backend/src/test/java/heartbeat/controller/pipeline/dto/response/PipelineTransformerTest.java

```diff
@@ -25,14 +25,39 @@ void shouldTransformToPipeline() {
 
 		Pipeline pipeline = PipelineTransformer.fromBuildKitePipelineDto(dto, orgId, orgName);
 
-		assertEquals(pipeline.id, "SLUG");
-		assertEquals(pipeline.name, "Name");
-		assertEquals(pipeline.repository, "repository");
-		assertEquals(pipeline.orgId, orgId);
-		assertEquals(pipeline.orgName, orgName);
-		assertEquals(pipeline.steps.size(), 1);
-		assertEquals(pipeline.steps.get(0), "Name1");
+		assertEquals("SLUG", pipeline.id);
+		assertEquals("Name", pipeline.name);
+		assertEquals("repository", pipeline.repository);
+		assertEquals(orgId, pipeline.orgId);
+		assertEquals(orgName, pipeline.orgName);
+		assertEquals("repository", pipeline.repoName);
+		assertEquals(1, pipeline.steps.size());
+		assertEquals("Name1", pipeline.steps.get(0));
+	}
+
+	@Test
+	void shouldTransformToPipelineWhenRepositoryContainsSlash() {
+		String orgId = "ORG_ID";
+		String orgName = "ORG_NAME";
+		List<StepsDTO> steps = List.of(StepsDTO.builder().name("Name1").build(), StepsDTO.builder().name("").build(),
+				StepsDTO.builder().build());
+		BuildKitePipelineDTO dto = BuildKitePipelineDTO.builder()
+			.slug("SLUG")
+			.name("Name")
+			.repository("organization/repository")
+			.steps(steps)
+			.build();
+
+		Pipeline pipeline = PipelineTransformer.fromBuildKitePipelineDto(dto, orgId, orgName);
 
+		assertEquals("SLUG", pipeline.id);
+		assertEquals("Name", pipeline.name);
+		assertEquals("organization/repository", pipeline.repository);
```

Comment:
same as line 58

Reply:
It's not same, and repository is ssh or http url.  Now I have modified the test to mock the GithubUtil static method